soc/sifive/sifive_freedom/CMakeLists.txt

```diff
@@ -2,4 +2,4 @@
 # SPDX-License-Identifier: Apache-2.0
 
 add_subdirectory(common)
-add_subdirectory(${SOC_SERIES})
+add_subdirectory(${CONFIG_SOC_SERIES})
```

Comment:
this should be `add_subdirectory(${SOC_SERIES})`, if SOC_SERIES is not equal to CONFIG_SOC_SERIES then that's a problem

Reply:
Hopefully they are always in sync:
https://github.com/zephyrproject-rtos/zephyr/blob/6e0945a82ab0a66728b442e37f7fc9e990e1d00f/cmake/modules/soc_v2.cmake#L26

iirc the `SOC_SERIES` version originates from a time when arch and soc had to be passed to the build, and thus not came through Kconfig.